### ISSUE TITLE
Poetry does not remove old dist-info folders

### ISSUE THREAD

**Issue by BD103** [python-poetry/poetry#5061]
- [ ] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] ~~If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).~~

- **OS version and name**: Ubuntu 20.04.2 LTS
- **Poetry version**: 1.1.8
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: [Click me](https://gist.github.com/BD103/f8a4f89019f8d521c0932f9e39e4218d)

## Issue

Hello! I am experiencing an error where Poetry does not remove old `pypackage-version.dist-info` folders. Normally this would not be an issue, but it can make importlib return an incorrect version number.

### Recreate this

Go to an empty folder, and do the following.

```shell
mkdir -p src/pypackage
poetry init -n --name pypackage
touch poetry.toml src/pypackage/__init__.py src/pypackage/__main__.py
```

For debugging purposes, I make Poetry create an environment in the current directory. Paste the following into `poetry.toml`:

```toml
[virtualenvs]
create = true
in-project = true
```

Paste the following into `src/pypackage/__main__.py`:

```python
from importlib.metadata import version

print(version("pypackage"))
```

Now run:

```shell
poetry lock
poetry install
```

Confirm that a `.venv` folder has been created, and that it contains `.venv/lib/python3.VERSION/site-packages/pypackage-0.1.0.dist-info`. Also confirm that the command works by running `poetry run python -m pypackage`. It should print `0.1.0`.

Now change the version number in `pyproject.toml` or run `poetry version patch`, then run `poetry install` again.

Notice how in the `site-packages` folder, both `pypackage-0.1.0.dist-info` and `pypackage-0.1.1.dist-info` exist. Run `poetry run python -m pypackage` again.

It will still print `0.1.0`, even though a later version is installed.

### Solution

My solution is to delete all `dist-info` folders matching the regex of the package's name, then installing it again. This will make importlib report the correct version. There are most likely better solutions, but this one would work.

### Workaround

If this issue hasn't been fixed, then the "solution" that I found was to delete the `.venv` folder and re-lock / install.

**Comment by finswimmer** [comment]
Hello @BD103,

Hello @jean-helsinki,

this was reported earlier in https://github.com/python-poetry/poetry/issues/2855 and is fixed in the current preview release (1.2.0a2).

fin swimmer

**Comment by BD103** [comment]
Thank you!
